### ISSUE TITLE
Cannot authenticate to private registry during the installation process 

### ISSUE THREAD

**Issue by DisaPadla** [pnpm/pnpm#5970]
It doesn't work only on macOS. I tried to reproduce issue on windows and it works fine.

Try to do `pnpm import` or `pnpm i`
### Code to reproduce the issue:
Code inside .npmrc
```
//artifactory.{name}.xyz/artifactory/api/npm/npm/:_authToken=[token]
always-auth=true
registry=https://artifactory.{name}.xyz/artifactory/api/npm/npm/
```
### Expected behavior:
Should install all packages

### Actual behavior:
```
 ERR_PNPM_FETCH_401  GET https://artifactory.{NAME}.xyz:443/artifactory/api/npm/npm/{SCOPE}/platform-jest-6.36.0.tgz: Unauthorized - 401
```
### Additional information:

 - `node -v` prints: v16.13.1
 - Windows, macOS, or Linux?: macOS Ventura 13.0.1

**Comment by ppatel31** [comment]
get same issue which is working with NPM & YARN but not working with PNPM

 ERR_PNPM_FETCH_401  GET https://name-of-the-components-******.d.codeartifact.us-east-1.amazonaws.com:443/npm/name-of-the-components/@my-scope/name-of-the-components/-/name-of-the-components-1.1.1.tgz: Unauthorized - 401

No authorization header was set for the request.

These authorization settings were found:
@my-scope:registry=https://name-of-the-components-******.d.codeartifact.us-east-1.amazonaws.com/npm/name-of-the-components/
//name-of-the-components-*******.d.codeartifact.us-east-1.amazonaws.com/npm/name-of-the-components/:_authToken=eyJ2[hidden]

**Comment by sunboshan** [comment]
When always-auth=true is ignored per 7.14.0 release https://github.com/pnpm/pnpm/releases/tag/v7.14.0

```
Ignore the always-auth setting.

pnpm will never reuse the registry auth token for requesting the package tarball, if the package tarball is hosted on a different domain.
```

how do I authenticate using private registry?

**Comment by sunboshan** [comment]
Found a workaround, is to add :443 to registry in .npmrc.

```
//artifactory.{name}.xyz/artifactory/api/npm/npm/:_authToken=[token]
always-auth=true
registry=https://artifactory.{name}.xyz/artifactory/api/npm/npm/
```

became

```
//artifactory.{name}.xyz:443/artifactory/api/npm/npm/:_authToken=[token]
always-auth=true
registry=https://artifactory.{name}.xyz:443/artifactory/api/npm/npm/
```

**Comment by yoavniran** [comment]
@sunboshan thank you, thank you!!! I wasted an hour now not understanding why the hell pnpm isnt installing from my AWS code artifactory. Your workaround just saved me hours more of frustration (& probably more hair-loss)!

Im on pnpm 7.29.0 (latest) and the workaround is required still!

**Comment by tlehtimaki** [comment]
> Found a workaround, is to add `:443` to registry in `~/.npmrc`.
> 
> ```
> //artifactory.{name}.xyz/artifactory/api/npm/npm/:_authToken=[token]
> always-auth=true
> registry=https://artifactory.{name}.xyz/artifactory/api/npm/npm/
> ```
> 
> became
> 
> ```
> //artifactory.{name}.xyz:443/artifactory/api/npm/npm/:_authToken=[token]
> always-auth=true
> registry=https://artifactory.{name}.xyz:443/artifactory/api/npm/npm/
> ```

Also stumbled upon the same issue today with version 7.27.1 and I think with version 7.9.3, but I only needed to add the `:443` with extra line for `_authToken`. I could not remove the original line, without it this did not work.

**Comment by hilja** [comment]
Let me add my 2¢. I had insane issues getting the artifactory auth working.

- Remove your project level `.npmrc` (if you have one)
- Create an empty `~/.npmrc`
- Run `pnpm login --registry https://artifactory.example.com/artifactory/api/npm/npm/` (or whatever that is for you)
- Fill in your username, password, and email and let the script populate the `.npmrc`

🙏

**Comment by mikeymop** [comment]
Hello, I was also experiencing this issue and was staying on `pnpm@7.6.0` in the meantime.

Wanted to follow up, today we tried `pnpm@8.0.0` and everything appears to be workin smoothly again, definitely give it a try if you were also staying back. :)

**Comment by dominikg** [comment]
Just ran into this on a private azure registry.

Tried the suggested :443 workaround and also adding more lenient scopes, still receiving a 401 response.

```shell
# .npmrc
//pkgs.dev.azure.com:443/<redacted>/_packaging/<redacted>/npm/registry/:username="${XXX}"
//pkgs.dev.azure.com:443/<redacted>/_packaging/<redacted>/npm/registry/:_password="${YYY}"
//pkgs.dev.azure.com:443/<redacted>/_packaging/<redacted>/npm/registry/:always-auth=true
//pkgs.dev.azure.com/<redacted>/_packaging/<redacted>/npm/registry/:username="${XXX}"
//pkgs.dev.azure.com/<redacted>/_packaging/<redacted>/npm/registry/:_password="${YYY}"
//pkgs.dev.azure.com/<redacted>/_packaging/<redacted>/npm/registry/:always-auth=true
//pkgs.dev.azure.com:443/<redacted>/:username="${XXX}"
//pkgs.dev.azure.com:443/<redacted>/:_password="${YYY}"
//pkgs.dev.azure.com:443/<redacted>/:always-auth=true
//pkgs.dev.azure.com/<redacted>/:username="${XXX}"
//pkgs.dev.azure.com/<redacted>/:_password="${YYY}"
//pkgs.dev.azure.com/<redacted>/:always-auth=true
registry=https://pkgs.dev.azure.com:443/<redacted>/_packaging/<redacted>/npm/registry/
engine-strict=true
```


```shell
# error output from pnpm install

 ERR_PNPM_FETCH_401  GET https://pkgs.dev.azure.com/<redacted>/_packaging/<redacted>/npm/registry/some-dependency: Unauthorized - 401

This error happened while installing a direct dependency of /path/to/pnpm-workspace/some-scope/some-package

An authorization header was used: Basic <redacted>

These authorization settings were found:
//pkgs.dev.azure.com:443/<redacted>/_packaging/<redacted>/npm/registry/:username=<redacted>
//pkgs.dev.azure.com:443/<redacted>/_packaging/<redacted>/npm/registry/:_password=[hidden]
//pkgs.dev.azure.com/<redacted>/_packaging/<redacted>/npm/registry/:username=<redacted>
//pkgs.dev.azure.com/<redacted>/_packaging/<redacted>/npm/registry/:_password=[hidden]
//pkgs.dev.azure.com:443/<redacted>/:username=<redacted>
//pkgs.dev.azure.com:443/<redacted>/:_password=[hidden]
//pkgs.dev.azure.com/<redacted>/:username=<redacted>
//pkgs.dev.azure.com/<redacted>/:_password=[hidden]
_password=[hidden]
username=<redacted>
```

Notes:
- XXX and YYY in .npmrc refer to env vars that are exported properly. the redacted output of :username in the error message shows they are being resolved.
- install works fine with `npm` and `pnpm@7.11.0`.
- install fails with latest pnpm 7 and pnpm 8

After installation with npm or older pnpm, the tarball urls in the lockfile do match the url patterns for the found authorization patterns.

Is there any way to better debug this or has someone else run into this and can share workarounds/a solution?

**Comment by nihauc12** [comment]
Moving from yarn to pnpm and encountered the same issue, this is what solved it for me:

this single line used to work with yarn:
`//registry.{domain}.com/:_authToken={token}`

I  had to add the 2nd line to make it work during pnpm import:
```
//registry.{domain}.com/:_authToken={token}
@{package-vendor}:registry=https://registry.{domain}.com:443/
```

**Comment by jakeparis** [comment]
Running into this same issue. Works fine for me on my local machine, but in GitLab CI I get 404. Also, as others said, if I simply do the CI job with npm instead, it works perfectly fine. 

## .gitlab-ci.yml
```yml
  ...
  before_script:
    - corepack enable
    - corepack prepare pnpm@latest --activate
    - pnpm config set store-dir .pnpm-store
    - pnpm config set //gitlab.com/api/v4/packages/npm/:_authToken $REGISTRY_ACCESS_TOKEN
    - pnpm config set @jakeparis:registry https://gitlab.com/api/v4/packages/npm/
  script: 
    - pnpm install
    - pnpm run build
    ...
```
And here's what I'm getting in the job logs:
```
$ pnpm config set store-dir .pnpm-store
$ pnpm config set //gitlab.com/api/v4/packages/npm/:_authToken $REGISTRY_ACCESS_TOKEN
$ pnpm config set @jakeparis:registry https://gitlab.com/api/v4/packages/npm/
$ pnpm install
Lockfile is up to date, resolution step is skipped
Progress: resolved 1, reused 0, downloaded 0, added 0
Packages: +2
++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: /builds/bates-ils/projects/wordpress/test-only-jparis/.pnpm-store/v3
  Virtual store is at:             node_modules/.pnpm
 WARN  GET https://gitlab.com/api/v4/projects/12345678/packages/npm/@jakeparis/wp-js-components/-/@jakeparis/wp-js-components-6.0.2.tgz error (ERR_PNPM_FETCH_404). Will retry in 10 seconds. 2 retries left.
Progress: resolved 2, reused 0, downloaded 1, added 1
 WARN  GET https://gitlab.com/api/v4/projects/12345678/packages/npm/@jakeparis/wp-js-components/-/@jakeparis/wp-js-components-6.0.2.tgz error (ERR_PNPM_FETCH_404). Will retry in 1 minute. 1 retries left.
 ERR_PNPM_FETCH_404  GET https://gitlab.com/api/v4/projects/12345678/packages/npm/@jakeparis/wp-js-components/-/@jakeparis/wp-js-components-6.0.2.tgz: Not Found - 404
No authorization header was set for the request.
```
In addition to using `pnpm config set` at runtime, I've also tried including a .npmrc file in the repo instead, and using `npm config set`, but neither one makes anything better. I've also triple-checked the token, and tried a new token. Neither helped. 

Based on the final error message **No authorization header was set for the request.**, it seems that pnpm is not even using the token I've set.

**Comment by giuliano-abinbev** [comment]
any update on this? i'm also experiencing issues with this

**Comment by giuliano-abinbev** [comment]
I also tried with different keys like `_auth`, `_authToken`, etc and still getting the same error. Looks like it grabs the correct key because I see `Bearer` when I use `_authToken` and `Basic` when I use `_auth`

**Comment by zkochan** [comment]
A PR has been submitted that should hopefully fix this: https://github.com/pnpm/pnpm/pull/7337

**Comment by alessandrojcm** [comment]
> A PR has been submitted that should hopefully fix this: #7337

Any ETA on when this will be released? Currently facing this issue building a Docker image in a Gitlab Runner

**Comment by zkochan** [comment]
It is released in 8.11.0
